Dockerfile

```diff
@@ -30,11 +32,13 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update -y && \
       curl
 
 RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
-    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-4.2.12-Linux-x86_64.sh && \
-    /bin/bash /Miniconda3-4.2.12-Linux-x86_64.sh -b -p /opt/conda && \
-    rm Miniconda3-4.2.12-Linux-x86_64.sh && \
-    /opt/conda/bin/conda install --yes conda==4.2.12 && \
-    echo "conda ==4.2.12" > /opt/conda/conda-meta/pinned
+    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-${OUR_CONDA_VERSION}-Linux-x86_64.sh && \
+    /bin/bash /Miniconda3-${OUR_CONDA_VERSION}-Linux-x86_64.sh -b -p /opt/conda && \
+    rm Miniconda3-${OUR_CONDA_VERSION}-Linux-x86_64.sh && \
+    /opt/conda/bin/conda install --yes conda==${OUR_CONDA_VERSION} && \
+    echo "conda ==${OUR_CONDA_VERSION}" > /opt/conda/conda-meta/pinned && \
+    conda install python==${OUR_PYTHON_VERSION} && \
+    echo "\npython ==${OUR_PYTHON_VERSION}" >> /opt/conda/conda-meta/pinned
```

Comment:
So the issue with the last PR (#10) was that the miniconda 4.3.11 will install python 3.6. Then when you try and update the root env, it balked, because it had python 3.5.2 listed. So pinning python made it resolve the deps properly to keep the root env at the right version of python.